frontends/common/resolveReferences/resolveReferences.cpp

```diff
@@ -16,74 +16,57 @@ limitations under the License.
 
 #include "resolveReferences.h"
 
-#include <sstream>
-
-#include <boost/range/adaptor/reversed.hpp>
-
-#include "frontends/common/options.h"
+#include "frontends/common/parser_options.h"
 
 namespace P4 {
 
-static const std::vector<const IR::IDeclaration *> empty;
-
 ResolutionContext::ResolutionContext() { anyOrder = P4CContext::get().options().isv1(); }
 
-std::vector<const IR::IDeclaration *> *ResolutionContext::memoizeDeclarations(
+const std::vector<const IR::IDeclaration *> &ResolutionContext::memoizeDeclarations(
     const IR::INamespace *ns) const {
-    auto decls = ns->getDeclarations();
+    std::vector<const IR::IDeclaration *> decls;
     if (auto nest = ns->to<IR::INestedNamespace>()) {
-        // boost::adaptors::reverse expects a reference
-        auto temp = nest->getNestedNamespaces();
-        for (auto nn : boost::adaptors::reverse(temp)) decls = nn->getDeclarations()->concat(decls);
+        for (const auto *nn : nest->getNestedNamespaces()) {
+            auto nnDecls = nn->getDeclarations();
+            // FIXME: should just insert()
+            for (const auto *decl : *nnDecls) decls.push_back(decl);
```

Comment:
Because the way how `Enumerator` is implemented.... `Enumerator::begin() / Enumerator::end()` return so-called `EnumeratorHandle` which does not satisfy iterator requirements (no trait types like `reference` / `value_type`, etc.)